examples/go-client/main.go

```diff
@@ -18,26 +18,62 @@ package main
 
 import (
 	"context"
-	"fmt"
 	"log"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types"
+
+	"github.com/gravitational/trace"
+	"github.com/pborman/uuid"
 )
 
 func main() {
+	ctx := context.Background()
 	log.Printf("Starting Teleport client...")
-	client, err := connectClient()
+
+	creds, err := client.ProfileCreds()
+	if err != nil {
+		log.Fatalf("Failed to read credentials: %v", err)
+	}
+
+	clt, err := client.New(client.Config{
+		Addrs: []string{"proxy.example.com:3025"},
+		Creds: creds,
+	})
 	if err != nil {
 		log.Fatalf("Failed to create client: %v", err)
 	}
-	defer client.Close()
+	defer clt.Close()
 
-	ctx := context.Background()
+	demoClient(ctx, clt)
+}
 
-	fmt.Println("")
-	roleCRUD(ctx, client)
+func demoClient(ctx context.Context, clt *client.Client) (err error) {
+	// Create a new access request for api-admin to use the admin role.
+	accessReq, err := types.NewAccessRequest(uuid.New(), "access-admin", "admin")
```

Comment:
nit: one reason `NewAccessRequest` might fail to validate is when no roles are used which is very easy to trip on. I'd suggest to change the signature to accommodate for that requirement by making a single role mandatory with the rest optional:
```go
// NewAccessRequest assembled an AccessRequest resource.
func NewAccessRequest(name, user, role string, additionalRoles ...string) (AccessRequest, error) {

```

Reply:
That's a good point. I plan on coming back to these `New` functions to make them more consistent, user-friendly, and free of business logic, so I'll keep this comment in mind.